server/registration/services/RegistrationService.js

```diff
@@ -115,6 +115,112 @@ module.exports = {
         }
       );
     }
+  },
+
+  /**
+   * Confirm registration
+   *
+   * @param {object} request - The request that was sent from the controller
+   */
+  confirmRegistration: async function confirmRegistration(request) {
+
+    let registration = null;
+
+    try {
+      registration = await Verification.findOne(
+        {
+          msisdn: request.msisdn
+        }
+      );
+
+      // check if there is an existing registration code
+      if (appUtil.isNullOrUndefined(registration)) {
+        return internalEventEmitter.emit(
+          registrationChannels.Internal.ConfirmRegistrationCompletedEvent,
+          {
+            statusCode: 400,
+            body: '' // send empty?
```

Comment:
Here we return a validation error like this:
```
var validationError = require('../../libs/error/ValidationError');
new validationError(
            'Some validation errors occurred.',
            [
              {
                code: errors.User.NUMBER_ALREADY_EXISTS,
                message: `A user with number [${request.msisdn}] already exists.`,
                path: ['msisdn'] // This is the property on that violated the validation rule
              }
            ]
          )
```

If it is a once off validation, otherwise you can create a validation class like `UserValidator` (with or without chaining) so that you can call it from different places. there are examples of this in `UserService`